dill/session.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+#
```

Comment:
### Exclude objects that fail to pickle

Discussion history:

> **mmakcerns:** Could you not create a filter function that excludes session objects that fail to serialize?
> 
> **leogama:** Do you mean excluding registered unpickleable types or actual objects that fail to serialize? My guess is that's no way to know if a particular object is pickable in advance, which means that creating a common filter for it, as the ones implemented so far, is infeasible. Maybe it can be implemented efficiently in `save_module_dict()` by writing the `dict` opcodes manually and trapping values that fail to pickle —and saving a `dill.UnpicklabeObject` singleton in treir place?
> 
> **mmakcerns:** There is a pretty extensive, albeit incomplete, list in `dill._objects`. One could, with relative ease, filter out all objects from the namespace that are listed as unpickleable. That's not what I was talking about, however... I meant "fails to pickle". I'm guessing this could be done by first iterating through the global namespace, and separating the items that pickle from the objects that don't. There are some tools for this in `dill.detect`, but essentially it is something like: try to pickle the object and when it fails, trap the exception and roll back appropriately. I even gave a simple filter to this effect in one of the old issues:
> ```python
> list(map(globals().pop, tuple(i for (i,j) in globals().items() if not dill.pickles(j) and i not in ('__builtins__',))))
> ```

Reply:
Moved to #527